matlab/JSBSim_SFunction.cpp

```diff
@@ -0,0 +1,620 @@
+    ssSetDWorkDataType(  S, 0, SS_DOUBLE); /* use SS_DOUBLE if needed */
+
+    ssSetDWorkWidth(     S, 1, ssGetNumDiscStates(S));//Work vector for states * may need to add actuator states!
+    ssSetDWorkDataType(  S, 1, SS_DOUBLE);
+
+	ssSetDWorkWidth(     S, 2, ssGetNumDiscStates(S));	//Work vector derivatives
+    ssSetDWorkDataType(  S, 2, SS_DOUBLE);
+
+	ssSetDWorkWidth(     S, 3, ssGetOutputPortWidth(S,1));//Work vector for flight controls outputs
+    ssSetDWorkDataType(  S, 3, SS_DOUBLE);
+
+	ssSetDWorkWidth(     S, 4, ssGetOutputPortWidth(S,2));//Work vector for propulsion outputs
+    ssSetDWorkDataType(  S, 4, SS_DOUBLE);
+
+	ssSetDWorkWidth(     S, 5, ssGetOutputPortWidth(S,3));//Work vector for calculated outputs
+    ssSetDWorkDataType(  S, 5, SS_DOUBLE);	
+
+    //ssSetDWorkWidth(     S, 6, ssGetInputPortWidth(S,1));//Work vector for input port for HW 
+    //ssSetDWorkDataType(  S, 6, SS_DOUBLE); /* use SS_DOUBLE if needed */
+    
+    //ssSetDWorkWidth(     S, 7, ssGetOutputPortWidth(S,4));//Work vector for output port for HW 
+    //ssSetDWorkDataType(  S, 7, SS_DOUBLE); /* use SS_DOUBLE if needed */
+
+    
+	ssSetNumPWork(S, 1); // reserve element in the pointers vector
+                         // to store a C++ object
+
+    ssSetNumNonsampledZCs(S, 0);
+
+    ssSetOptions(S, 0);
+		
+}
+
+
+
+/* Function: mdlInitializeSampleTimes =========================================
+ * Abstract:
+ *    This function is used to specify the sample time(s) for your
+ *    S-function. You must register the same number of sample times as
+ *    specified in ssSetNumSampleTimes.
+ */
+static void mdlInitializeSampleTimes(SimStruct *S)
+{
+    //ssSetSampleTime(S, 0, CONTINUOUS_SAMPLE_TIME);
+    ssSetSampleTime(S, 0, delta_t);
+    ssSetOffsetTime(S, 0, 0.0);
+
+}
+
+#define MDL_INITIALIZE_CONDITIONS   /* Change to #undef to remove function */
+#if defined(MDL_INITIALIZE_CONDITIONS)
+  /* Function: mdlInitializeConditions ========================================
+   * Abstract:
+   *    In this function, you should initialize the continuous and discrete
+   *    states for your S-function block.  The initial states are placed
+   *    in the state vector, ssGetContStates(S) or ssGetRealDiscStates(S).
+   *    You can also perform any other initialization activities that your
+   *    S-function may require. Note, this routine will be called at the
+   *    start of simulation and if it is present in an enabled subsystem
+   *    configured to reset states, it will be call when the enabled subsystem
+   *    restarts execution to reset the states.
+   */
+  static void mdlInitializeConditions(SimStruct *S)
+  {	
+	    /* Create new JSBSimInterface object and initialize it with delta_t and
+		   also create a pointer to the JSBSimInterface object so we can access its member
+		   functions.
+		*/
+	    ssGetPWork(S)[0] = (void *) new JSBSimInterface(&FDMExec, delta_t);
```

Comment:
```suggestion
	    ssGetPWork(S)[0] = (void *) new JSBSimInterface(delta_t);
```
Make `FDMExec` a member of `JSBSimInterface`. It is not used anywhere else in the code `JSBSim_SFunction.cpp` anyway.